### PR TITLE
[PLATFORM-3258] Use RustTLS in favor of OpenSSL

### PR DESCRIPTION
**Pull request by bytekast**

### CODE REVIEW

**Review by anowell — COMMENTED** [review]
I'm arguing in my own head whether it's worth exposing a rustls and nativetls as 2 separate features for this client given the lack-of-industry-battle-testing for rustls or if that isn't really a big deal for a client, and it's perfectly fine to just hard-code rustls.

But I do feel like it'd only be a couple extra lines though to support both:

```
 reqwest = {version = "0.9.19", default_features = false}

[features]
default = ["native-tls"]
native-tls = ["reqwest", "reqwest/native-tls"]
rustls-tls = ["reqwest", "reqwest/rustls-tls"]
```

and then you'd need an extra line to configure the `Client` depending on which TLS implementation was chosen, e.g. https://doc.rust-lang.org/std/macro.cfg.html

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 /target
 /Cargo.lock
 /src/version.rs
+.idea/
```
Comment (zeryx) [inline .gitignore:5]: another jetbrains fan I see :+1:

**Comment by bytekast** [comment]
@anowell this doesn't seem to work
```
            inner_client: if cfg!(feature == "rustls-tls") {
                Arc::new(Client::builder().use_rustls_tls().build().unwrap())
            } else {
                Arc::new(Client::new())
            },
```

rusttls is already an optional feature in reqwest. could we use the same flags used there?

**Review by anowell — APPROVED** [review]
This looks good to me

**Pull request merged**